enhancements/network-observability.md

```diff
@@ -0,0 +1,367 @@
+The total number of bytes and packets are just estimations when sampling
+is used.  Therefore if you need full visibility of the traffic and accuracy,
+the only choice is not do sampling.
+
+3. Resources required<br>
+    Exporting data is potentially CPU, memory and bandwidth-intensive, and
+typically requires a large amount of storage.  The CPU/memory/bandwidth
+resources can be managed by the sampling data rate.  The storage requirements
+can be controlled by the data retention policy or how long to keep the data.
+
+    Nevertheless, in order to maintain the same performance that you have
+prior to enabling network observability, will require additional resources.
+
+
+### Risks and Mitigations
+
+1. Resource Risk<br>
+    Not having enough resources (e.g. CPU) can be disruptive to the system,
+and care must be taken to not jeopardize the health of the network by enabling
+network observability.
+
+2. Privacy<br>
+    Operators need to be aware of what data is being collected as it might
+expose customers to privacy issues.  In sensitive cases, storing IP addresses
+is problematic as it can reveal the user's general location and can be used
+to find out other information about the user, including possibly the user's
+identity.
+
+3. Cost<br>
+    It may not be clear how much additional cost the customer will incur,
+particularly if this is managed by Red Hat.  Additional nodes may need to
+be added to avoid the resource risk described above.
+
+4. Data accuracy<br>
+    If data sampling is turned on, then data accuracy cannot be relied on
+so using this for audit reports or criminal investigation are not viable.
+
+
+## Design Details
+
+### Software Architecture
+This is the software architecture at a high level.
+
+![architectural diagram](arch22.drawio.svg)
+
+The components with hatch marks indicate work that needs to be done.
+
+For the first release, it intentionally leaves out the message broker,
+the plan being Kafka, for simplicity.  For the specific choice of
+component, separate enhancements will provide the detail analysis on
+these decision.
+
+
+### Software Components
+
+#### OVS
+Open vSwitch (OVS) is the NetFlow exporter in the OpenShift cluster.  When
+Network Observability is enabled, each OVS in the node will be configured to
+send IPFIX data.  This is an all-or-none proposition, meaning there will
+not be an option to selectively choose some OVS to be exporters only.  There
+will be a configuration parameter to set the sampling rate.  Other parameters
+such as the cache size may also be exposed.
+
+#### Flow Collector
+The Flow Collector is responsible for receiving and normalizing the data
+sent by the NetFlow exporter.  This component is [GoFlow 2](https://github.com/netsampler/goflow2).
+The enhancement is at [Enable flows collection](https://github.com/mariomac/enhancements/blob/enable-flows-collection/enhancements/network/enable-flows-collection.md).
```

Comment:
Ideally this would link to either a merged enhancement or a PR against the enhancements repo.

Reply:
Waiting for the PR...

Reply:
There's actually an open PR: https://github.com/openshift/enhancements/pull/850
However we're currently working on re-scoping that enhancement proposal to something more focused on the work related to OVN, because it's what we are committed to do alongside 4.10. I'm working on it [here](https://github.com/mariomac/enhancements/pull/2).